### ISSUE TITLE
Expected behavior of an IEventStore?

### ISSUE THREAD

**Issue by aaronincincy** [gautema/CQRSlite#82]
Looking at the built in Repository it looks like after it saves the events to the event store it passes them on to the event publisher.  Looking at the InMemoryEventStore in the sample, it looks like it saves the events to the in memory database, then calls the event publisher again.  This seems like it would publish events twice unless I'm missing something.  When implementing my own IEventStore should I reproduce the same behavior?

**Comment by gautema** [comment]
Great to hear that you found out about this one. The problem with using the repository to publish events is that transactions are required for it to work. This creates the need for two phase commits in many systems. To avoid this, it's better that the event store handles publishing.

**Comment by aaronincincy** [comment]
Yeah, I've since changed my approach again and I'm using an Azure Function
with the CosmosDB trigger to publish events.  I think this should solve the
problem you mentioned with transactions.  Thanks for following up, I'm
really enjoying CqrsLite!
